src/main/java/org/carlmontrobotics/subsystems/Drivetrain.java

```diff
@@ -292,16 +313,17 @@ public void periodic() {
             // module.move(0, goal);
         }
 
-        field.setRobotPose(odometry.getPoseMeters());
+        // field.setRobotPose(odometry.getPoseMeters());
+
+        field.setRobotPose(poseEstimator.getEstimatedPosition());
 
+        // odometry.update(gyro.getRotation2d(), getModulePositions());
 
-        odometry.update(gyro.getRotation2d(), getModulePositions());
+        poseEstimator.update(gyro.getRotation2d(), getModulePositions());
         //odometry.update(Rotation2d.fromDegrees(getHeading()), getModulePositions());
 
+        updateMT2PoseEstimator();
 
-        //setPose(new Pose2d(SmartDashboard.getNumber("set x", getPose().getTranslation().getX()), SmartDashboard.getNumber("set y", getPose().getTranslation().getY()), Rotation2d.fromDegrees(getHeading())));
```

Comment:
The pose will only be set when the drivetrain is initialized. Don't you want it such that you can change the pose throughout testing without having to restart the robot?

Reply:
Btw, what Ryan meant by "have a specific event that causes the pose to be updated", is that only when the value changes should the pose actually be set. Otherwise, the pose will constantly be set to the pose on smartdashboard.